Lib/test/test_capi/test_object.py

```diff
@@ -251,15 +251,27 @@ def test_object_setattr_null_exc(self):
         class Obj:
             pass
         obj = Obj()
-
         obj.attr = 123
-        with self.assertRaises(SystemError):
-            _testcapi.object_setattr_null_exc(obj, 'attr')
-        self.assertTrue(hasattr(obj, 'attr'))
 
-        with self.assertRaises(SystemError):
-            _testcapi.object_setattrstring_null_exc(obj, 'attr')
-        self.assertTrue(hasattr(obj, 'attr'))
+        exc = ValueError("error")
+        with self.assertRaises(SystemError) as cm:
+            _testcapi.object_setattr_null_exc(obj, 'attr', exc)
+        self.assertIs(cm.exception.__context__, exc)
+        self.assertIsNone(cm.exception.__cause__)
+        self.assertHasAttr(obj, 'attr')
+
+        with self.assertRaises(SystemError) as cm:
+            _testcapi.object_setattrstring_null_exc(obj, 'attr', exc)
+        self.assertIs(cm.exception.__context__, exc)
+        self.assertIsNone(cm.exception.__cause__)
+        self.assertHasAttr(obj, 'attr')
+
+        with self.assertRaises(SystemError) as cm:
+            # undecodable name
+            _testcapi.object_setattrstring_null_exc(obj, b'\xff', exc)
+        self.assertIs(cm.exception.__context__, exc)
+        self.assertIsNone(cm.exception.__cause__)
+        self.assertHasAttr(obj, 'attr')
```

Comment:
This is not needed. We did nothing with attribute "attr".

Reply:
Calling SetAttr() with NULL value is supposed to delete the attribute.

Reply:
Oh I see, on this test we don't pass "attr". I removed the check.